pyuvsim/tests/test_telescope.py

```diff
@@ -211,3 +211,37 @@ def test_beamlist_errors(beam_objs):
     tel0 = pyuvsim.Telescope('tel0', array_location, newbeams)
     tel1 = pyuvsim.Telescope('tel1', array_location, beam_objs)
     assert tel0 != tel1
+
+
+def test_beamlist_consistency(beam_objs):
+    newbeams = copy.deepcopy(beam_objs)
+    
+    # Does check, but raises no error
+    pyuvsim.BeamList(newbeams[:2])
+
+    with pytest.raises(pyuvsim.BeamConsistencyError):
+        # Raises because analytic beams have no Nfeeds
+        pyuvsim.BeamList(newbeams[:3])
+
+def test_beamlist_consistency_properties(beam_objs):
+    newbeams = copy.deepcopy(beam_objs)
+    beamlist = pyuvsim.BeamList(newbeams[:2])
+    assert beamlist.x_orientation == newbeams[0].x_orientation
+
+def test_beamlist_consistency_stringmode(beam_objs):
+    newbeams = copy.deepcopy(beam_objs)
+    beamlist = pyuvsim.BeamList(newbeams[:2])
+    beamlist.set_str_mode()
+    beamlist.check_consistency(force=True)
+    assert beamlist.string_mode
+    with pytest.warns(UserWarning) as record:
+        beamlist.check_consistency(force=False)
+        if not record:
+            pytest.fail("Expected a warning!")
```

Comment:
This is an interesting approach, why are you using it instead of the `match` keyword to `pytest.warns`?. Another option that I like even better is to use pyuvdata.test.check_warnings to check warnings -- it's got the same pattern of usage as `pytest.warns`, but it checks that the warnings that are specified exactly match (in number, type and message) the warnings raised by the code.

Reply:
oh nice, didn't know that!